### DISCUSSION TITLE
Get posts from users I follow

### DISCUSSION THREAD

**Discussion by brielov** [prisma/prisma#4936] [Q&A]
I'm trying to make a feed populated with posts from users I follow. I have the following schema:

```prisma
model User {
  id String @id @default(cuid())
  following User[] @relation("follows")
  followedBy User[] @relation("follows")
  posts Post[]
}

model Post {
  id String @id @default(cuid())
  title String
  author User @relation(fields: [authorId], references: [id])
  authorId String
}
```

I tried using fluent api like this `prisma.user.findUnique({ where: { id: 'my own id' }}).following().posts()` but it only allow me untill the `following()` call.

How can I implement this feature?

Also, it would be nice to include my own posts in this feed, sorted by `publishedAt`

**Comment by ryands17** [comment]
Hey @brielov 👋 

In this case, you wouldn't be able to do this with the fluent API as the `following()` call will directly return the list of users following. You would need to do this with `include`: (more info [here](https://www.prisma.io/docs/concepts/components/prisma-client/field-selection#include))


```ts
  await prisma.user.findUnique({
    where: { id: 'something' },
    include: { following: { include: { posts: true } } },
  })
```

> Also, it would be nice to include my own posts in this feed, sorted by publishedAt

As for this, I don't think this is possible in the same query, so you would either have to make two queries (one for the User's posts and the other for the followers posts) or use [raw sql](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) via the `union` operator.

**Reply by brielov** [reply]
@ryands17  thank you so much for the reply. I ended up using a raw query something like this (I have no idea if this is performant, but It works)

```sql
SELECT
        *
        FROM
          "Post"
        WHERE
          "authorId" in(
            SELECT
              _follow. "A" AS A FROM _follow
            WHERE
              _follow. "B" = $1)
        OR
          "authorId" = $1
        ORDER BY
          "createdAt" DESC
        LIMIT 10  
```

**Reply by ryands17** [reply]
Yeah this should work as well :)

The same can be done with Prisma as well for a better experience:

```ts
  const following = await prisma.user.findFirst({
    where: { id: 'own-id' },
    select: { following: { select: { id: true } } },
  })

  const posts = await prisma.post.findMany({
    where: {
      author: {
        id: { in: [...following.following.map((user) => user.id), 'own-id'] },
      },
    },
    take: 10
  })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
